boa_engine/src/bytecompiler/jump_control.rs

```diff
@@ -0,0 +1,366 @@
+        self
+    }
+
+    pub(crate) const fn with_has_finally(mut self, value: bool) -> Self {
+        self.has_finally = value;
+        self
+    }
+
+    pub(crate) const fn with_for_of_in_loop(mut self, value: bool) -> Self {
+        self.for_of_in_loop = value;
+        self
+    }
+}
+
+// ---- `JumpControlInfo` const fn methods ---- //
+
+impl JumpControlInfo {
+    pub(crate) const fn label(&self) -> Option<Sym> {
+        self.label
+    }
+
+    pub(crate) const fn start_address(&self) -> u32 {
+        self.start_address
+    }
+
+    pub(crate) const fn kind(&self) -> JumpControlInfoKind {
+        self.kind
+    }
+
+    pub(crate) const fn in_catch(&self) -> bool {
+        self.in_catch
+    }
+
+    pub(crate) const fn has_finally(&self) -> bool {
+        self.has_finally
+    }
+
+    pub(crate) const fn finally_start(&self) -> Option<Label> {
+        self.finally_start
+    }
+
+    pub(crate) const fn for_of_in_loop(&self) -> bool {
+        self.for_of_in_loop
+    }
+
+    pub(crate) const fn decl_envs(&self) -> u32 {
+        self.decl_envs
+    }
+}
+
+impl JumpControlInfo {
+    /// Sets the `label` field of `JumpControlInfo`.
+    pub(crate) fn set_label(&mut self, label: Option<Sym>) {
+        assert!(self.label.is_none());
+        self.label = label;
+    }
+
+    /// Sets the `start_address` field of `JumpControlInfo`.
+    pub(crate) fn set_start_address(&mut self, start_address: u32) {
+        self.start_address = start_address;
+    }
+
+    /// Sets the `in_catch` field of `JumpControlInfo`.
+    pub(crate) fn set_in_catch(&mut self, value: bool) {
+        self.in_catch = value;
+    }
+
+    /// Sets the `finally_start` field of `JumpControlInfo`.
+    pub(crate) fn set_finally_start(&mut self, label: Label) {
+        self.finally_start = Some(label);
+    }
+
+    /// Increments the `decl_env` field of `JumpControlInfo`.
+    pub(crate) fn inc_decl_envs(&mut self) {
+        self.decl_envs += 1;
+    }
+
+    /// Decrements the `decl_env` field of `JumpControlInfo`.
+    pub(crate) fn dec_decl_envs(&mut self) {
+        self.decl_envs -= 1;
+    }
+
+    /// Pushes a `Label` onto the `break` vector of `JumpControlInfo`.
+    pub(crate) fn push_break_label(&mut self, break_label: Label) {
+        self.breaks.push(break_label);
+    }
+
+    /// Pushes a `Label` onto the `try_continues` vector of `JumpControlInfo`.
+    pub(crate) fn push_try_continue_label(&mut self, try_continue_label: Label) {
+        self.try_continues.push(try_continue_label);
+    }
+}
+
+// `JumpControlInfo` related methods that are implemented on `ByteCompiler`.
+impl ByteCompiler<'_, '_> {
+    /// Pushes a generic `JumpControlInfo` onto `ByteCompiler`
+    ///
+    /// Default `JumpControlInfoKind` is `JumpControlInfoKind::Loop`
+    pub(crate) fn push_new_jump_control(&mut self) {
+        self.jump_info.push(JumpControlInfo::default());
+    }
+
+    pub(crate) fn set_jump_control_label(&mut self, label: Option<Sym>) {
```

Comment:
To avoid having a lot of duplicated setters between `ByteCompiler` and `JumpControlInfo`, what about having `ByteCompiler::current_jump_control_mut` returning `&mut JumpControlInfo`?